rewrite-java/src/main/java/org/openrewrite/java/format/SpacesVisitor.java

```diff
@@ -53,29 +54,41 @@ public SpacesVisitor(SpacesStyle style, @Nullable EmptyForInitializerPadStyle em
     }
 
     <T extends J> T spaceBefore(T j, boolean spaceBefore) {
-        if (spaceBefore && StringUtils.isNullOrEmpty(j.getPrefix().getWhitespace())) {
+        if (!j.getComments().isEmpty()) {
+            return j;
+        }
```

Comment:
Does IntelliJ take this approach when you have it format code with comments in it?

Reply:
when turning on `BeforeParentheses.MethiodDeclarationParentheses` and `WithIn.MethodDeclarationParenthese` in Intellij.
The behavior is 
```java
private int testParams    /*comment*/    (    /*comment*/    int x    /*comment*/    ) {
    return x;
}
```
is turned into
```java
private int testParams    /*comment*/ (    /*comment*/    int x    /*comment*/ ) {
    return x;
}
```
so it just changed the suffix of the comment accordingly. I think I can make the same change.

Reply:
addressed. now it behaves the same as IntelliJ now.